Slim/Control/Request.pm

```diff
@@ -2398,16 +2398,19 @@ sub __matchingRequest {
 # return compiled regexp representing the $possibleNames array of arrays
 sub __requestRE {
 	my $possibleNames = shift || return qr /./;
-	my $regexp = '';
-
-	my $i = 0;
+	my @regexp = ();
 
 	for my $names (@$possibleNames) {
-		$regexp .= ',' if $i++;
-		$regexp .= (scalar @$names > 1) ? '(?:' . join('|', @$names) . ')' : $names->[0];
+
+		# Bracket each name using word boundaries to avoid
+		# "play" matching "play", "playlist", and "display".
+
+		push @regexp,
+			'(?:' . join('|', map { "\\b$_\\b" } @$names) . ')'
+			if scalar @$names;
 	}
 
-	return qr /$regexp/;
+	return qr /${\(join(',', @regexp))}/;
```

Comment:
Please, just for readability, as regexes can be confusing enough themselves: could you store the result of the join in a variable `$regexp` and return as before?